### PR TITLE
Address cargo fmt and clippy issues

### PR DESCRIPTION
**Pull request by dongsupark**
Address issues of `cargo fmt` as well as `cargo clippy`.

In CI, run also `cargo fmt` and `cargo test`.
Use `actions-rs/cargo` Github Actions.

### CODE REVIEW

**Comment by dongsupark** [comment]
(ignore the unit test failures, as it will be fixed in the other PR.)

**Review by krnowak — APPROVED** [review]

File: .rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# Disable the granular width settings automatically configured following max_width.
+use_small_heuristics="Off"
+
+# Maximum width of each line.
+max_width=200
+
+# Maximum width of each function call
+# It is needed as use_small_heuristics=Off.
+fn_call_width=80
```
Comment (pothos) [inline .rustfmt.toml]: Is the reformat really worth it or can we rather go with the defaults to prevent git changes for all files?

Comment (pothos) [reply .rustfmt.toml]: `cargo fmt` shows no changes for me, do have a custom global config that causes the changes?

**Review by pothos — CHANGES_REQUESTED** [review]
Let's not use a custom `.rustfmt.toml` but stick to the existing (default rustfmt) formatting

File: .rustfmt.toml (hunk 1 above)
Comment (dongsupark) [reply .rustfmt.toml]: Done with removing a local rustfmt.toml.

**Comment by dongsupark** [comment]
Done with removing a local rustfmt.toml.

**Review by krnowak — APPROVED** [review]

**Review by pothos — APPROVED** [review]

**Pull request merged**